src/Elasticsearch/Connections/Connection.php

```diff
@@ -168,6 +168,11 @@ public function __construct(
         }
         $port = $hostDetails['port'];
 
+        if (isset($connectionParams['client']['port_in_header']) && $connectionParams['client']['port_in_header']) {
+            if (!in_array((int) $port, [80,443])) {
+                $host .= ":$port";
```

Comment:
Instead of appending the port here in the constructor, I think you should append when a request is made in `performRequest()`. The setting is called `include_port_in_host_header`, so it makes sense to only include it in the header, and the header is only constructed in `performRequest`

Reply:
(If you do it in `performRequest()` instead it is possible to get just the host part when you call `getHost()`. Those that need to get the port as well have a `getPort()` method for that. But the way its implemented right now makes it impossible to get the original `host` part back from the `Connection` object.)

Reply:
Very good point! I'm refactoring it as you suggested. Thanks!